.travis.yml

```diff
@@ -3,11 +3,25 @@ language: python
 matrix:
   include:
     - os: linux
-      python: 3.5
+      python: '3.6'
       env: DEPLOY_ARGS="-3 -b"
     - os: linux
-      python: 2.7
+      python: '2.7'
       env: DEPLOY_ARGS="-2 -b"
+    - env: TEST='pytest'
+      python: '3.6'
+      install: pip install --upgrade 'pip>=9' 'pytest>=3,<4' 'flake8>=3,<4'
```

Comment:
Can we use a `requirements.txt` file so that we don't duplicate these versions and dependencies in three places?

Reply:
The reason I didn't is that `flake8` isn't required to run `pytest|python setup.py test`.  For a long time, github was shipping an old `pip` version in their `2.7` env that was choking on version constraints between certain pypi packages.  I also vaguely recall there was some sort dep resolution problem with an old version of `pytest` being preinstalled -- so I cut'n'pasted from a sqre pypi package.

Reply:
I've added a `.travis.requirements.txt` and testing if I can dry up `.travis.yaml` with yaml anchors.

Reply:
I've managed to condense it down to:
```
    - &pytest
      env: TEST='pytest'
      python: '3.6'
      install: pip install --upgrade -r .travis.requirements.txt
      script: |
        python setup.py test
        # https://github.com/tholo/pytest-flake8/issues/27
        flake8 bin/lsst-build
    - <<: *pytest
      python: '2.7'
```